cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -67,33 +88,54 @@ func (rm *AzureResourceManager) GetDeploymentResourceOperations(
 	}
 
 	if strings.TrimSpace(resourceGroupName) == "" {
-		return resourceOperations, nil
+		return topLevelDeploymentOperations, nil
+	}
+
+	allLevelsDeploymentOperations := []*armresources.DeploymentOperation{}
+
+	resourceIdPrefix := fmt.Sprintf("/subscriptions/%s/resourceGroups/%s", subscriptionId, resourceGroupName)
+	isWithinResourceGroup := func(operation *armresources.DeploymentOperation) bool {
+		return strings.HasPrefix(*operation.Properties.TargetResource.ID, resourceIdPrefix)
 	}
 
 	// Find all resource group deployments within the subscription operations
 	// Recursively append any resource group deployments that are found
-	deploymentOperations := make(map[string]*armresources.DeploymentOperation)
-	for _, operation := range resourceOperations {
-		if operation.Properties.TargetResource != nil &&
-			*operation.Properties.TargetResource.ResourceType == string(AzureResourceTypeDeployment) &&
-			*operation.Properties.ProvisioningOperation == armresources.ProvisioningOperationCreate {
+	innerLevelDeploymentOperations := make(map[string]*armresources.DeploymentOperation)
+	for _, operation := range topLevelDeploymentOperations {
+		if !hasTargetResourceData(operation) {
+			// Operations w/o target data can't be resolved. Ignoring them
+			continue
+		}
+		if !isWithinResourceGroup(operation) {
+			// topLevelDeploymentOperations might include deployments NOT within the resource group which we don't want to
+			// resolve
+			continue
+		}
```

Comment:
Did we intentionally change the behavior slightly? I believe this function used to return all operations regardless of whether it is "within resource group". Personally, I think the function should return all non-deployment operations that target resources. I think we should codify this in the documentation.

Reply:
It also may be odd that the inner deployment cares about Timestamps, whereas the outer does not. It should be consistent...

Reply:
We were getting all deploy-ops (top level) from a deployment.
Then, we would be discovering all deploy-ops within a specific `resource group`.  But the strategy for getting all deploy-ops within the `resource group` was:
- For each top-level deploy-ops which is a `deployment type` -> DEP
  - Get deploy-ops from `resource group` with id => DEP

This means that, trying to get  `deploy-ops` from `resource group` for a `DEP` which is not part of that `resource group` would just return `404 not found`.

So, I am changing the strategy to filter `deployment-ops` on top level, which are related to the `resource group`.  So we would not be getting the `404 not found` error.

Reply:
@vhvb1989 I'm referring to the returned result (i.e., return signature `([]*armresources.DeploymentOperation, error`)) not about deployment expansion. There needs to be a description and consistency of what deployment operations we actually return. I believe the intention is to return "resource-level operations". A resource level operation could? exist at the subscription level. We're now filtering it out immediately on line 109.

Reply:
I see.
Yes, we were returning `all deployment-ops` from the top level (regardless of the type [ `deployment` or `resource` ] ) *plus* `all deployment-ops` from the inner levels, but only the ones of type `resource`.

So, anyone calling this function would need to check the `resource type` while walking through each element.  I changed it to return only resource type = `resource`, as that's what the consumer (the reporting component) is expecting,

I will add docs to the method.